consensus/src/quorum_store/batch_store.rs

```diff
@@ -146,18 +147,73 @@ impl BatchStore {
             persist_subscribers: DashMap::new(),
             expiration_buffer_usecs,
         };
-        let db_content = db_clone
-            .get_all_batches()
-            .expect("failed to read data from db");
+
+        if is_new_epoch {
+            tokio::task::spawn_blocking(move || {
+                Self::gc_previous_epoch_batches_from_db(db_clone, epoch);
+            });
+        } else {
+            Self::populate_cache_and_gc_expired_batches(
```

Comment:
@bchocho @zekun000 I split the QS part of the PR out and addressed your comments here. I renamed the method to be clear.